### PR TITLE
Added a few resources, including DCPL and some tutorials

### PR DESCRIPTION
**Pull request by xuanruiqi**
Added DCPL book, lcc compiler, and Erlang BEAM among others

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -214,10 +216,12 @@ A curated list of awesome resources, learning materials, tools, frameworks, plat
 ### Tutorials
 
   * [A Tutorial Implementation of a Dependently Typed Lambda Calculus](https://www.andres-loeh.de/LambdaPi/).
+  * [Algorithm W Step By Step](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.65.7733&rep=rep1&type=pdf).
```
Comment (aalhour) [inline README.md]: Can you move this `Algorithm W Step By Step` resource to the Papers section, since it is a research paper?

File: README.md [hunk 2]
```diff
@@ -87,6 +87,8 @@ A curated list of awesome resources, learning materials, tools, frameworks, plat
   * [Advanced Topics in Types and Programming Languages, B. Pierce](https://www.amazon.com/dp/0262162288).
   * [Building an Optimizing Compiler, B. Morgan](https://www.amazon.com/dp/155558179X).
   * [Compiling with Continuations, A. Appel](https://www.amazon.com/dp/052103311X).
+  * [Design Concepts in Programming Languages, F. Turbak, D. Gifford, M. Sheldon](https://www.amazon.com/dp/0262201755) - also contains some introductory level material.
+  * [The Garbage Collection Handbook: The Art of Automatic Memory Management, R. Jones, A. Hosking, E. Moss](https://www.amazon.com/dp/1420082795).
```
Comment (aalhour) [inline README.md]: Can you move this book further down in the list right before `The SSA Book` resource to maintain the alphabetical order?

File: README.md [hunk 3]
```diff
@@ -372,6 +377,7 @@ _This section aims at listing code projects of Compilers, Interpreters, Translat
 
 ## Runtimes and VMs
 
+  * [Erlang BEAM](https://github.com/erlang/otp).
```
Comment (aalhour) [inline README.md]: Can you reorder this list to maintain the alphabetical order of resources?

File: README.md (hunk 1 above)
Comment (xuanruiqi) [inline README.md]: It's a tutorial that takes a paper format; it doesn't present any research. Therefore I believe it's a tutorial.

Comment (aalhour) [inline README.md]: Makes sense.

**Pull request merged**